libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -362,6 +362,8 @@ bool ASTJsonConverter::visit(FunctionDefinition const& _node)
 		make_pair("implemented", _node.isImplemented()),
 		make_pair("scope", idOrNull(_node.scope()))
 	};
+	if (_node.isPartOfExternalInterface())
+		attributes.emplace_back("functionSelector", formatNumber(_node.externalIdentifier()));
```

Comment:
The function selector should also be added for public state variables.

Reply:
I've added this inside `ASTJsonConverter::visit(VariableDefinition const& _node)`:

https://github.com/ethereum/solidity/blob/b9ac8da1e16f6a6c3c22934d4a70991cfaf434da/libsolidity/ast/ASTJsonConverter.cpp#L389-L390